components/script/dom/element.rs

```diff
@@ -458,7 +443,10 @@ impl Element {
     }
 
     pub fn is_shadow_host(&self) -> bool {
-        self.rare_data().as_ref().unwrap().shadow_root.is_some()
+        if let Some(rare_data) = self.rare_data().as_ref() {
```

Comment:
With a `shadow_root` accessor this can be `self.shadow_root().is_some()` which reads much nicer imo.